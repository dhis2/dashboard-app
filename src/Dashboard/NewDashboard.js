import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import DashboardVerticalOffset from './DashboardVerticalOffset';
import EditBar from '../ControlBarContainer/EditBar';
import DashboardContent from './DashboardContent';
import { acSetEditNewDashboard } from '../actions/editDashboard';

class NewDashboard extends Component {
    componentDidMount() {
        this.props.setNewDashboard();
    }

    render() {
        return (
            <Fragment>
                <EditBar />
                <DashboardVerticalOffset />
                <div className="dashboard-wrapper">
                    <DashboardContent editMode={true} />
                </div>
            </Fragment>
        );
    }
}

export default connect(
    null,
    {
        setNewDashboard: acSetEditNewDashboard,
    }
)(NewDashboard);
