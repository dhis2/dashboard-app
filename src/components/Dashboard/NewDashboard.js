import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';

import { acSetEditNewDashboard } from '../../actions/editDashboard';
import DashboardVerticalOffset from './DashboardVerticalOffset';
import EditBar from '../ControlBar/EditBar';
import DashboardContent from './DashboardContent';

class NewDashboard extends Component {
    componentDidMount() {
        this.props.setNewDashboard();
    }

    render() {
        return (
            <Fragment>
                <EditBar />
                <DashboardVerticalOffset editMode={true} />
                <div className="dashboard-wrapper">
                    <DashboardContent editMode={true} />
                </div>
            </Fragment>
        );
    }
}

export default connect(
    null,
    {
        setNewDashboard: acSetEditNewDashboard,
    }
)(NewDashboard);
