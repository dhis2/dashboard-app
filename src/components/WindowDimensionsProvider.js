import React, { createContext, useContext, useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import debounce from 'lodash/debounce'

export const WindowDimensionsCtx = createContext(null)

const windowDims = () => ({
    height: window.innerHeight,
    width: window.innerWidth,
})

const WindowDimensionsProvider = ({ children }) => {
    const [dimensions, setDimensions] = useState(windowDims())

    useEffect(() => {
        const handleResize = debounce(() => {
            setDimensions(windowDims())
        }, 200)
        window.addEventListener('resize', handleResize)
        return () => {
            window.removeEventListener('resize', handleResize)
        }
    }, [])

    return (
        <WindowDimensionsCtx.Provider value={dimensions}>
            {children}
        </WindowDimensionsCtx.Provider>
    )
}

WindowDimensionsProvider.propTypes = {
    children: PropTypes.node.isRequired,
}

export default WindowDimensionsProvider

export const useWindowDimensions = () => {
    return useContext(WindowDimensionsCtx)
}
