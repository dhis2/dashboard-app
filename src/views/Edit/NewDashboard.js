import React, { useEffect } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import i18n from '@dhis2/d2-i18n'
import cx from 'classnames'

import DashboardContainer from '../../components/DashboardContainer'
import ActionsBar from './ActionsBar'
import TitleBar from './TitleBar'
import ItemGrid from './ItemGrid'
import LayoutPrintPreview from '../print/PrintLayoutDashboard'
import NotSupportedNotice from './NotSupportedNotice'

import { acSetEditNewDashboard } from '../../actions/editDashboard'
import { sGetIsPrintPreviewView } from '../../reducers/editDashboard'

import classes from './styles/NewDashboard.module.css'

const NewDashboard = props => {
    useEffect(() => {
        props.setNewDashboard()
    }, [])

    return (
        <>
            <div
                className={cx(classes.container, 'dashboard-scroll-container')}
            >
                <ActionsBar />
                {props.isPrintPreviewView ? (
                    <LayoutPrintPreview fromEdit={true} />
                ) : (
                    <DashboardContainer>
                        <TitleBar />
                        <ItemGrid />
                    </DashboardContainer>
                )}
            </div>
            <div className={classes.notice}>
                <NotSupportedNotice
                    className={classes.notSupportedNotice}
                    message={i18n.t(
                        'Creating dashboards on small screens is not supported. Resize your screen to return to create mode.'
                    )}
                />
            </div>
        </>
    )
}

NewDashboard.propTypes = {
    isPrintPreviewView: PropTypes.bool,
    setNewDashboard: PropTypes.func,
}

const mapStateToProps = state => ({
    isPrintPreviewView: sGetIsPrintPreviewView(state),
})

export default connect(mapStateToProps, {
    setNewDashboard: acSetEditNewDashboard,
})(NewDashboard)
