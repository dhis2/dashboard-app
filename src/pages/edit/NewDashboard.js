import i18n from '@dhis2/d2-i18n'
import cx from 'classnames'
import PropTypes from 'prop-types'
import React, { useState, useEffect } from 'react'
import { connect } from 'react-redux'
import { Redirect } from 'react-router-dom'
import { acSetEditNewDashboard } from '../../actions/editDashboard'
import DashboardContainer from '../../components/DashboardContainer'
import Notice from '../../components/Notice'
import { useWindowDimensions } from '../../components/WindowDimensionsProvider'
import { setHeaderbarVisible } from '../../modules/setHeaderbarVisible'
import { isSmallScreen } from '../../modules/smallScreen'
import { sGetIsPrintPreviewView } from '../../reducers/editDashboard'
import LayoutPrintPreview from '../print/PrintLayoutDashboard'
import ActionsBar from './ActionsBar'
import ItemGrid from './ItemGrid'
import classes from './styles/NewDashboard.module.css'
import TitleBar from './TitleBar'

const NewDashboard = props => {
    const { width } = useWindowDimensions()
    const [redirectUrl, setRedirectUrl] = useState(null)

    useEffect(() => {
        if (isSmallScreen(width)) {
            setRedirectUrl('/')
            return
        }
        setHeaderbarVisible(true)
        props.setNewDashboard()
    }, [])

    if (redirectUrl) {
        return <Redirect to={redirectUrl} />
    }

    return (
        <>
            <div
                className={cx(classes.container, 'dashboard-scroll-container')}
            >
                <ActionsBar />
                {props.isPrintPreviewView ? (
                    <LayoutPrintPreview fromEdit={true} />
                ) : (
                    <DashboardContainer>
                        <TitleBar />
                        <ItemGrid />
                    </DashboardContainer>
                )}
            </div>
            <div className={classes.notice}>
                <Notice
                    title={i18n.t('Not supported')}
                    message={i18n.t(
                        'Creating dashboards on small screens is not supported. Resize your screen to return to create mode.'
                    )}
                />
            </div>
        </>
    )
}

NewDashboard.propTypes = {
    isPrintPreviewView: PropTypes.bool,
    setNewDashboard: PropTypes.func,
}

const mapStateToProps = state => ({
    isPrintPreviewView: sGetIsPrintPreviewView(state),
})

export default connect(mapStateToProps, {
    setNewDashboard: acSetEditNewDashboard,
})(NewDashboard)
