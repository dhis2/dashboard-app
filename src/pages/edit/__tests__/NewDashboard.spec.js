import { render } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import configureMockStore from 'redux-mock-store'
import WindowDimensionsProvider from '../../../components/WindowDimensionsProvider.js'
import NewDashboard from '../NewDashboard.js'

/* eslint-disable react/prop-types */
jest.mock('@dhis2/ui', () => {
    const originalModule = jest.requireActual('@dhis2/ui')

    return {
        __esModule: true,
        ...originalModule,
        NoticeBox: function Mock({ warning, ...props }) {
            return (
                <div
                    className="ui-NoticeBox"
                    data-warning={warning}
                    {...props}
                />
            )
        },
        CenteredContent: function Mock({ children }) {
            return <div className="ui-CenteredContent">{children}</div>
        },
    }
})
/* eslint-enable react/prop-types */

jest.mock(
    '../ActionsBar',
    () =>
        function MockActionsBar() {
            return <div>ActionsBar</div>
        }
)

jest.mock(
    '../TitleBar',
    () =>
        function MockTitleBar() {
            return <div>TitleBar</div>
        }
)
jest.mock(
    '../ItemGrid',
    () =>
        function MockEditItemGrid() {
            return <div>ItemGrid</div>
        }
)

jest.mock(
    '../../print/PrintLayoutDashboard',
    () =>
        function MockLayoutPrintPreview() {
            return <div>LayoutPrintPreview</div>
        }
)

const mockStore = configureMockStore()

const store = {
    editDashboard: {
        id: '',
        access: { update: true, delete: true },
        printPreviewView: false,
    },
}

test('NewDashboard renders dashboard', () => {
    const { container } = render(
        <>
            <header style={{ height: '48px' }} />
            <Provider store={mockStore(store)}>
                <WindowDimensionsProvider>
                    <NewDashboard />
                </WindowDimensionsProvider>
            </Provider>
        </>
    )

    expect(container).toMatchSnapshot()
})

test('NewDashboard renders print preview', () => {
    store.editDashboard.printPreviewView = true

    const { container } = render(
        <>
            <header style={{ height: '48px' }} />
            <Provider store={mockStore(store)}>
                <WindowDimensionsProvider>
                    <NewDashboard />
                </WindowDimensionsProvider>
            </Provider>
        </>
    )
    expect(container).toMatchSnapshot()
})
