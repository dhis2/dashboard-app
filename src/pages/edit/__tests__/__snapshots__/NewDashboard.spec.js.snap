// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewDashboard renders dashboard 1`] = `
<div>
  <header
    style="height: 48px;"
  />
  <div
    class="container dashboard-scroll-container"
  >
    <div>
      ActionsBar
    </div>
    <div
      class="container dashboard-scroll-container"
      data-test="inner-scroll-container"
    >
      <div>
        TitleBar
      </div>
      <div>
        ItemGrid
      </div>
    </div>
  </div>
  <div
    class="notice"
  >
    <div
      class="jsx-498096601 center top"
      data-test="dhis2-uicore-centeredcontent"
    >
      <div
        class="jsx-498096601 center-inner-content"
      >
        <div
          class="noticeContainer"
        >
          <div
            class="jsx-3561221191 root warning"
            data-test="dhis2-uicore-noticebox"
          >
            <div
              class="jsx-3231412310"
              data-test="dhis2-uicore-noticebox-icon"
            >
              <svg
                color="#e56408"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12.847 2.794l.056.102 8.416 17.674a1 1 0 01-.786 1.423l-.117.007H3.584a1 1 0 01-.947-1.322l.044-.108 8.416-17.674a1 1 0 011.75-.102zM12 18a1 1 0 100 2 1 1 0 000-2zm1-9h-2v7h2z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <div
              class="jsx-3561221191"
            >
              <h6
                class="jsx-1234954849"
                data-test="dhis2-uicore-noticebox-title"
              >
                Not supported
              </h6>
              <div
                class="jsx-3638294246"
                data-test="dhis2-uicore-noticebox-message"
              >
                Editing dashboards on small screens is not supported. Resize your screen to return to edit mode.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`NewDashboard renders print preview 1`] = `
<div>
  <header
    style="height: 48px;"
  />
  <div
    class="container dashboard-scroll-container"
  >
    <div>
      ActionsBar
    </div>
    <div>
      LayoutPrintPreview
    </div>
  </div>
  <div
    class="notice"
  >
    <div
      class="jsx-498096601 center top"
      data-test="dhis2-uicore-centeredcontent"
    >
      <div
        class="jsx-498096601 center-inner-content"
      >
        <div
          class="noticeContainer"
        >
          <div
            class="jsx-3561221191 root warning"
            data-test="dhis2-uicore-noticebox"
          >
            <div
              class="jsx-3231412310"
              data-test="dhis2-uicore-noticebox-icon"
            >
              <svg
                color="#e56408"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12.847 2.794l.056.102 8.416 17.674a1 1 0 01-.786 1.423l-.117.007H3.584a1 1 0 01-.947-1.322l.044-.108 8.416-17.674a1 1 0 011.75-.102zM12 18a1 1 0 100 2 1 1 0 000-2zm1-9h-2v7h2z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <div
              class="jsx-3561221191"
            >
              <h6
                class="jsx-1234954849"
                data-test="dhis2-uicore-noticebox-title"
              >
                Not supported
              </h6>
              <div
                class="jsx-3638294246"
                data-test="dhis2-uicore-noticebox-message"
              >
                Editing dashboards on small screens is not supported. Resize your screen to return to edit mode.
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
